client/app/bundles/course/duplication/pages/Duplication/__test__/ObjectDuplication.test.jsx

```diff
@@ -40,7 +39,7 @@ describe('<ObjectDuplication />', () => {
 
     await waitFor(() => expect(spy).toHaveBeenCalled());
 
-    const data = useAppSelector(selectDuplicationStore);
+    const data = store.getState().duplication;
```

Comment:
What's the reason we move to `.tsx` then regress back to `.jsx`?

Reply:
oh sorry, this one I thought I can also convert from jsx to tsx, but apparently doing so creates more problem (and hence I also understand why the other jstest uses jsx rather than tsx) and hence I convert it back to what it's used to be..

Reply:
> create more problem

what problem?

Reply:
sorry.. the conversion towards tsx previously failed because I used the wrong function (I should maintained `store.getState().duplication` instead of using the other one, to ensure that the content of the store is updated when the duplication value is fetched)..

However, do note that at this point, the type of `data` is not defined yet since duplicationStore is still in js format (reformatting it to ts will take too much time, and hence I am not keen to do that at this point). Hence, I just re-cast `data` into the types that is needed for this testing, that is `destinationCourses` and `materialComponents`.

Reply:
Don't leave migration to TS half-done. How much time is too much time?